src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java

```diff
@@ -789,9 +790,25 @@ private int updateStopTimesForPatternStops(List<PatternStop> patternStops) throw
         for (String tripId : timesForTripIds.keySet()) {
             // Initialize travel time with previous stop time value.
             int cumulativeTravelTime = timesForTripIds.get(tripId);
+            int timepointNumber = 0;
+            double timepointSpeed = 0;
+            double previousShapeDistTraveled = 0;
+            PatternStop currentTimepoint = patternStops.get(0); // First stop must be a timepoint
             for (PatternStop patternStop : patternStops) {
+                boolean isTimepoint = patternStop.timepoint == 1;
+                // If we have reached a timepoint (which is not the last), we calculate the speed between it and the next timepoint.
+                if (isTimepoint && interpolateStopTimes && timepointNumber < timepoints.size()-1){
+                    timepointNumber++;
+                    previousShapeDistTraveled = currentTimepoint.shape_dist_traveled;
+                    PatternStop nextTimePoint = timepoints.get(timepointNumber);
+                    timepointSpeed = (nextTimePoint.shape_dist_traveled - currentTimepoint.shape_dist_traveled) / nextTimePoint.default_travel_time;
```

Comment:
Should we check if `default_travel_time` can't be zero before doing this division?

Reply:
Good point, yeah I'll add a case for that